src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -27,37 +28,39 @@
 import java.util.Map;
 import java.util.TimeZone;
 import java.util.concurrent.*;
+import java.util.function.Function;
 
-import ai.rapids.cudf.ColumnVector;
-import ai.rapids.cudf.DType;
-import ai.rapids.cudf.HostColumnVector;
-import ai.rapids.cudf.Table;
+import ai.rapids.cudf.*;
 
 public class GpuTimeZoneDB {
 
   public static final int TIMEOUT_SECS = 300;
 
-
   // For the timezone database, we store the transitions in a ColumnVector that is a list of 
   // structs. The type of this column vector is:
-  //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32>>
+  //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32, looseInstant: int64>>
```

Comment:
```java
/* In order to do binary search on segments:
 *   [year, month, day, hour, minute, seconds, mocroseconds]
 * Alternations are:
 * 1. Save the segments for each transition, this will introduce more columns/sub-columns.
 * 2. Before search on segments, treat segments as UTC time and convert segments to UTC timestamp.
 *    This will require kernel to handle like: convert segments to UTC timestamp. It will like:
 *      LocalDate.toEpochDay() * 86400 + LocalTime.toSecondOfDay + LocalTime.toNanoOfDay
 * 3. Above 2 methods both are complicated, below introduced a more simple way: loose instant method
*/

```

Reply:
Please ignore the previous comment.
Actually we need `looseInstant`. We use this column to do binary search.
Or we have to add other columns represent local date time like Java `ZoneRules`
```
    for (ZoneOffsetTransition trans : transitionList) {
            if (trans.isGap()) {
                localTransitionList.add(trans.getDateTimeBefore());
                localTransitionList.add(trans.getDateTimeAfter());
            } else {
                localTransitionList.add(trans.getDateTimeAfter());
                localTransitionList.add(trans.getDateTimeBefore());
            }
            localTransitionOffsetList.add(trans.getOffsetAfter());
        }
```

Reply:
Okay so all I want, then is some comments explaining what looseInstant is and ideally how it can be used. This is not for me right now, this is for anyone who comes in the future and needs to deal with a gap transition in some way.